core/src/test/kotlin/com/strumenta/kolasu/model/IndexingTest.kt

```diff
@@ -0,0 +1,80 @@
+package com.strumenta.kolasu.model
+
+import org.junit.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+
+class IndexingTest {
+
+    @Test
+    fun computeIdsWithDefaultWalkerAndIdProvider() {
```

Comment:
I think it would be useful to have a test showing the IdProvider that assigns IDs only to Node which are referred. When used in this case, I think that b1 should not have an ID

Reply:
This should be covered now